src/App.tsx

```diff
@@ -15,36 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type SortType = 'length' | 'abc' | '';
+
+const getSortedGoods = (sortType: SortType, reversed: boolean): string[] => {
+  let sortedGoods = [...goodsFromServer];
+
+  if (sortType) {
+    sortedGoods.sort((a, b) => {
+      switch (sortType) {
+        case 'abc':
+          return a.localeCompare(b);
+        case 'length':
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    sortedGoods = sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState<SortType>('');
+  const [reversed, setReversed] = useState<boolean>(false);
+
+  const visibleGoods = getSortedGoods(sortType, reversed);
+
+  const handleSort = (type: SortType): void => {
+    setSortType(type);
+  };
+
+  const handleReverse = (): void => {
+    setReversed(!reversed);
+  };
```

Comment:
[CODE STYLE] - The function's name `handleReverse` should start with a verb that describes what the function does. Consider renaming it to `toggleReversed` to indicate that it's toggling the reversed state.